packages/react-query/src/bin/postinstall.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { getTanStackReactQueryPackageJson } from './utils/package'
+import { switchVersion } from './utils/switchVersion'
+
+vi.mock('./utils/package')
+vi.mock('./utils/switchVersion')
+
+describe('postinstall', () => {
+  const mockConsoleWarn = vi.spyOn(console, 'warn')
+  const mockGetPackageJson = vi.mocked(getTanStackReactQueryPackageJson)
+  const mockSwitchVersion = vi.mocked(switchVersion)
```

Comment:
Long but easy to understand is better in my opinion! because we use too many package.json to name it as mockGetPackageJson

```suggestion
  const mockConsoleWarn = vi.spyOn(console, 'warn')
  const mockGetTanStackReactQueryPackageJson = vi.mocked(getTanStackReactQueryPackageJson)
  const mockSwitchVersion = vi.mocked(switchVersion)
```